### PR TITLE
[BH-1444][BH-1475] Fix vertical layout

### PR DESCRIPTION
**Pull request by mjan-dev**
**Description**

This PR fixes issues with vertical layouts:
- wrong hour displayed for 12 in 24 format,
- scrambled chars without stable alignment, variable with hours.

Note:
This will be cherry-picked to master.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

**Review by pawel-mudita — APPROVED** [review]

File: products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp [hunk 1]
```diff
@@ -23,54 +23,77 @@ namespace gui
...
+        hourSecond->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Center));
         hourSecond->setEdges(RectangleEdge::None);
         hourSecond->setText("0");
-        hourSecond->setMargins(getHourMargins(fontName));
+        hourSecond->drawUnderline(false);
 
-        hBoxHours->addWidget(hourFirst);
-        hBoxHours->addWidget(hourSecond);
+        hBoxHoursLeft->addWidget(hourFirst);
+        hBoxHoursRight->addWidget(hourSecond);
```
Comment (paweljonskim) [inline products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp:59]: is adding those widgets neccessary? `hourFirst` and `hourSecond` are constructed with parents `hBoxHoursLeft` and `hBoxHoursRight` already. Maybe we can omit those?

**Review by paweljonskim — APPROVED** [review]

File: products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp (hunk 1 above)
Comment (mjan-dev) [reply products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp]: Removed. Good point

File: products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp [hunk 2]
```diff
@@ -23,54 +23,77 @@ namespace gui
...
-        minuteSecond->setAlignment(Alignment(Alignment::Horizontal::Right, Alignment::Vertical::Top));
+        minuteSecond->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Top));
         minuteSecond->setEdges(RectangleEdge::None);
         minuteSecond->setText("0");
         minuteSecond->drawUnderline(false);
 
-        hBoxMinutes->addWidget(minuteFirst);
-        hBoxMinutes->addWidget(minuteSecond);
+        hBoxMinutesLeft->addWidget(minuteFirst);
+        hBoxMinutesRight->addWidget(minuteSecond);
```
[the remark this answers is not in the document]
Comment (mjan-dev) [reply products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp]: Removed. Good point

File: products/BellHybrid/apps/common/include/common/widgets/ClockVertical.hpp [hunk 3]
```diff
@@ -17,8 +17,8 @@ namespace style::time_set_spinner_vertical
         inline constexpr auto size = 6U;
     } // namespace focus
 
-    inline constexpr auto small_margin = 6U;
-    inline constexpr auto big_margin   = 44U;
+    inline constexpr auto small_margin    = 6U;
+    inline constexpr auto big_margin      = 44U;
     inline constexpr auto colossal_margin = -40;
```
Comment (paweljonskim) [inline products/BellHybrid/apps/common/include/common/widgets/ClockVertical.hpp:22]: I think those might be no longer needed as they were used in spaceMarginsMap which is deleted

Comment (mjan-dev) [reply products/BellHybrid/apps/common/include/common/widgets/ClockVertical.hpp:22]: Removed

**Comment by paweljonskim** [comment]
Selection_509
LGTM

**Pull request merged**